components/bridge/AddToken.tsx

```diff
@@ -0,0 +1,204 @@
+import React, { useEffect, useState } from "react";
+import airdrop from "@/assets/airdrop.svg";
+import { AnimatePresence, motion } from "framer-motion";
+import cross from "@/assets/cross.svg";
+import copy from "@/assets/copy.svg";
+import { selectToastSlice, toggleAddTokenToast } from "@/store/toastSlice";
+import { useAppDispatch, useAppSelector } from "@/store/store";
+import { appConfig } from "@/lib/config";
+import tick from "@/public/tick.png";
+import Image from "next/image";
+
+const AddToken = () => {
+  const tokenSlice = useAppSelector(selectToastSlice);
+  const dispatch = useAppDispatch();
+  const [checked, setChecked] = useState(false);
+  useEffect(() => {
+    window.addEventListener("click", (e) => {
+      if ((e.target as HTMLElement).id === "modal-bg") {
+        dispatch(toggleAddTokenToast(false));
+      }
+    });
+  }, []);
+  useEffect(() => {
+    setChecked(false);
+  }, [tokenSlice.token]);
+  const getTokenAddress = (token: string) => {
+    const tokenDetails = appConfig.wrappedBridge.fuse.tokens.find(
+      (t) => t.symbol === token
+    );
+    return tokenDetails?.address;
+  };
+  const getTokenDecimals = (token: string) => {
+    const tokenDetails = appConfig.wrappedBridge.fuse.tokens.find(
+      (t) => t.symbol === token
+    );
+    return tokenDetails?.decimals;
+  };
+  const getTokenIcon = (token: string) => {
```

Comment:
@MayankMittal1 to reduce computation and avoid redundant functions, you can create one function called `getTokenDetails`, instead of three: "getTokenAddress", "getTokenDecimals", and "getTokenIcon", then call its properties.

To get the "tokenDetails address", now you can use `getTokenDetails(tokenSlice.token as string).address`.